### PR TITLE
Language tweaks

### LINKED ISSUE #5481: Disable moderation security
RC1
The setting "Disable moderation security" In the tooltip which accompanies this setting it says,

> This disables the additional password check for the moderation page. **This is not recommended!**

Thing is, this appears to be disabled by default. 

Admin > Maintenance > Server Settings > Security >> Disable moderation security

### LINKED ISSUE #5621: Possible typo (and more) in english $helptxt['dont_show_attach_under_post']
#### Description
[(Original thread here.)](https://www.simplemachines.org/community/index.php?topic=567411.0)

##### 1/3
Wrong(?):
> $helptxt['dont_show_attach_under_post'] = 'If enabled, the attachments embedded ~~on~~ the post won\'t appear under the post but are still subject to be deleted.';

Suggested fix:
> $helptxt['dont_show_attach_under_post'] = 'If enabled, the attachments embedded _**in**_ the post won\'t appear under the post but are still subject to be deleted.';

###### 2/3
I'm not sure about the meaning of everything after the "but"; I guess the point is that all other "normal"-attachment-related settings still apply to them?  (Because that's what `Admin.english.php:$txt['dont_show_attach_under_post_sub']` says.)
Also, why even have the help text when it is basically a shortened repetiton of `Admin.english.php:$txt['dont_show_attach_under_post_sub']`? Shouldn't the extra help text be _more_ verbose by default?

##### 3/3
[Irisado pointed out](https://www.simplemachines.org/community/index.php?topic=567411.msg4017470#msg4017470) that there needs to be a comma before the "but" (my grammar skills aren't enough the comment on that; in German the comma is optional).

### Steps to reproduce
1. Read Help.english-utf8.php

### Environment (complete as necessary)
- Version/Git revision: Revision: #95312(?); are the language files in a Git repo somewhere?

### LINKED ISSUE #5622: $helptxt['themes']: Unclear reference
This is the help text of the Admin section "top header"; there are 4 menu items below, each of which could be selected when reading this help text, and in none of them do I find something about changing theme settings to the right side of the page. Or is it (clumsily) referring to the right-most menu item, "Modify Themes"?

### Steps to reproduce
1. Read Help.english-utf8.php

### Environment (complete as necessary)
- Version/Git revision: Revision: #95312

### LINKED ISSUE #5626: Question re: $helptxt['attachmentThumbWidth'] and -Height
#### Description
[Both texts](https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Themes/default/languages/Help.english.php#L285-L286) currently say (only showing `-Width`):
> `$helptxt['attachmentThumbWidth'] = 'Only used with the &quot;Resize images when showing under posts&quot; setting the maximum width to resize attachments down from. They will be resized proportionally.';`

I think this needs to be corrected to either something like this:
> $helptxt['attachmentThumbWidth'] = 'Only used with ~~the~~ &quot;Resize images when showing under posts&quot;\_,\_ setting the maximum width to resize attachments down ~~from~~ **to**. They will be resized proportionally.';

Or something like this:
> $helptxt['attachmentThumbWidth'] = 'Only used with the &quot;Resize images when showing under posts&quot; setting\_:\_ the maximum width to resize attachments down ~~from~~ **to**. They will be resized proportionally.';

Or is something lost in translation here?

### Steps to reproduce
1. Read `Help.english.php` or browse the admin center attachment settings

### Environment (complete as necessary)
- Version/Git revision: HEAD

### Additional information/references
[Original thread here.](https://www.simplemachines.org/community/index.php?topic=567406.0)

### LINKED ISSUE #5661: Collected language (and other) issues
Dumping my rather draft-y braindump-y collection of issues here, since [in #5634, Sesquipedalian asked me to do it.](https://github.com/SimpleMachines/SMF2.1/issues/5634#issuecomment-490558277)
I'm translating my own comments without context here, so they might not be clear (I'll improve them later, if necessary). I will roll this into a PR in a couple of days or weeks.

I can't figure out how to nest numbered lists; behold, this is gonna be a mess…

1. ​https://github.com/SimpleMachines/SMF2.1/issues/5621 **[2/3 fixes in #5778 (3rd maybe later)]**
1. ​https://github.com/SimpleMachines/SMF2.1/issues/5622 **[will eventually be fixed in #5785]**
1. ​https://github.com/SimpleMachines/SMF2.1/issues/5623 **[fix in #5778]**
1. ​https://github.com/SimpleMachines/SMF2.1/issues/5624 **[fix in #5778]**
1. ​https://github.com/SimpleMachines/SMF2.1/issues/5626 **[fix in #5778]**
1. ​https://github.com/SimpleMachines/SMF2.1/issues/5631 **[fix in #5778]**
1. ​https://github.com/SimpleMachines/SMF2.1/issues/5633 **[fix in #5778]**
1. ​https://github.com/SimpleMachines/SMF2.1/issues/5634 **[fix in #5778]**
1. ​https://github.com/SimpleMachines/SMF2.1/pull/5635 **[fix in #5778]**
1. ​https://github.com/SimpleMachines/SMF2.1/pull/5636 **[open]**
1. ​https://github.com/SimpleMachines/SMF2.1/pull/5637 **[open]**
1. ​https://github.com/SimpleMachines/SMF2.1/pull/5638  **[fix in #5778]**
1. some strings, f.e. in `Errors`, say stuff like "less than x seconds ago". Is this some weird kind of rounding down in the code, or just a mistake in the text? **[wontfix]**
1. `Errors[…].php`:
  * a) `too_many_groups`: "please remove some [from your selection]", otherwise might trigger over-eager users to delete the actual groups (don't make me have to think) **[fix in #5778]**
  * b) `post_upload_error`: "is can be caused" **[fix in #5778]**
  * c) `search_invalid_weights`: "should be configure to be" **[fix in #5778]**
  * d) `package_get_error_theme_no_new_version`: turn logic around **[(updated) fix in #5778]**
  * e) `profile_error_custom_field_mail_fail`: -"format" **[fix in #5778]**
  * f) `loadavg_userstats_disabled`: switch sentence order. **[fix in #5778]**
  * g) `mboards_parent_own_child_error`: change to "You can not make a board into a sub-board of one of its own sub-boards."(?) **[invalid/withdrawn]**
  * h) `s/Json/JSON/g` **[fix in #5778]**
15. there are many deprecated Headers (pre-`2.1rc2`). Is this language-dependent and needs to be fixed in the english files first, or can I update the German files independently? **[manual update on release (of rc3 or stable 2.1?)]**
16. `Install[…].php`:
  * a) `db_settings_prefix_info`: s/value/key or option/ **[fix in #5778]**
  * b) `db_populate_info2`: should directly use `install_settings_proceed` (if that assumed reference is correct), and generally all these String-Quote-References should (code-)use the original string; this is a defused version of my [suggestion in the "Department of Redundany Department" thread](https://www.simplemachines.org/community/index.php?topic=567249.0) for only verbatim quotes **[wontfix]**
  * c) `error_session_missing`: [wtf?](https://www.simplemachines.org/community/index.php?topic=567598.0) **[fix in #5778]**
    * Fix-attempt: `$txt['error_session_missing'] = 'The installer was unable to detect session support in your server\'s installation of PHP. Please ask your host to ensure that PHP was compiled with session support (which in fact is the PHP default, meaning your host currently has explicitly disabled it.)';`
 * d) `error_username_too_long`: `s/25/26/` or `'Username may only be up to 25 characters long.'` seems to more accurately reflect the code **[fix in #5778]**
 * e) `ftp_path_help` is unclear. It's missing the hint to the user that the path is relative, and the text seems to describe at least two different paths, the FTP main path and the SMF-Path. **[maybe later]**
 * f) `upgrade_areyouready`: use (code-)reference to the Continue-Variable. **[wontfix]**
 * g) `upgrade_backup_table` attempt a forum upgrade and check… something? (probably under what conditions this string is shown, or whether it is accurate, or what the exact meaning is so I can actually translate it well). **[fix in #5778]**
  * h) `upgrade_error_script_js`: Texts like this refer to Tools which probably have Deeplinks ([in this case, yes](https://www.simplemachines.org/community/index.php?topic=567632.msg4018741#msg4018741): https://download.simplemachines.org/?tools), but they just link to the main site. Is this on purpose (maybe because they might change over the years; but then they could still be grepped for and updated)? Why make the user search, this will probably keep users from taking this route. **[fix in #5778]**
  * i) `upgrade_time_ago_{hms,ms,s}`: -"ago" in the text **[fix in #5778]**
  * j) `upgrade_repondtime`: Variablename +s  **[fix in #5778]**
17. in the German translation, all the (non-compound) hyphens are invalid and need(ed; done locally but not uploaded everything) to be replaced by (German) [hyphens (En dash/Halbgeviertstrich _with_ surrounding spaces)](https://en.wikipedia.org/wiki/Dash#En_dash), I vaguely remember English requires(?) [hyphens (Em dash/Geviertstrich _without_ surrounding spaces)](https://en.wikipedia.org/wiki/Dash#Em_dash) too but I'm unsure; I have a few more details in my private draft of what I currently call the "SMF German [translation] Glossary", but I need to freshen up on the details. **[will probably have to fix it myself 😉 ]**

### PR DESCRIPTION
**Pull request by frandominguezl**
Closes #5621 
Closes #5622 
Closes #5626
Closes #5481

Ref #5623 
Ref #5624 
Ref #5631 
Ref #5633 
Ref #5634

And also closes #5661 addressing all the mentioned issues.
I want to personally thank @m4z for his delightful effort towards the language files. It's really appreciated.

This is a WIP for now since I have to check a few things before merging this. Nothing serious, I just want to use only a PR for this.

### CODE REVIEW

**Comment by sbulen** [comment]
FYI - I put "Has fix" labels on #5621, #5622, #5626, & #5661

**Comment by frandominguezl** [comment]
Unless I missed something this is ready to go.

**Comment by m4z** [comment]
I'm planning to add comments and do a quick review of the changes over the next few days (before Saturday morning).

**Comment by frandominguezl** [comment]
Sounds great, I'll mark this as WIP then. Thanks for your input.

File: Themes/default/languages/EmailTemplates.english.php [hunk 1]
```diff
@@ -200,7 +200,7 @@
 $txt['new_announcement_subject'] = 'New announcement: {TOPICSUBJECT}';
 $txt['new_announcement_body'] = '{MESSAGE}
 
-To unsubscribe from these announcements, login to the forum and uncheck "Receive forum announcements and important notifications by email." in your profile.
+To unsubscribe from these announcements, login to the forum and uncheck "Receive forum announcements and important notifications by email." in your profile <em>(Your username > Notificacions > General Settings)</em>.
```
Comment (m4z) [inline Themes/default/languages/EmailTemplates.english.php]:
```suggestion
To unsubscribe from these announcements, login to the forum and uncheck "Receive forum announcements and important notifications by email." in your profile <em>(Your username > Notifications > General Settings)</em>.
```

File: Themes/default/languages/Help.english.php [hunk 2]
```diff
@@ -356,9 +356,9 @@
...
 $helptxt['secureCookies'] = 'Enabling this setting will force the cookies created for users on your forum to be marked as secure. Only enable this setting if you are using HTTPS throughout your site, otherwise it will mean no-one will be able to log in properly!';
-$helptxt['securityDisable'] = 'This <em>disables</em> the additional password check for the administration page. This is not recommended!';
+$helptxt['securityDisable'] = 'When enabled, this <em>disables</em> the additional password check for the administration page. This is not recommended!';
 $helptxt['securityDisable_why'] = 'This is your current password. (the same one you use to login.)<br><br>The requirement to enter this helps ensure that you want to do whatever administration you are doing, and that it is <strong>you</strong> doing it.';
-$helptxt['securityDisable_moderate'] = 'This <em>disables</em> the additional password check for the moderation page. This is not recommended!';
+$helptxt['securityDisable_moderate'] = 'When enabled, his <em>disables</em> the additional password check for the moderation page. This is not recommended!';
```
Comment (m4z) [inline Themes/default/languages/Help.english.php]:
```suggestion
$helptxt['securityDisable_moderate'] = 'When enabled, this <em>disables</em> the additional password check for the moderation page. This is not recommended!';
```

File: Themes/default/languages/ModerationCenter.english.php [hunk 3]
```diff
@@ -172,6 +172,7 @@
 
 // Do not use numeric entries in the below string.
 $txt['mc_reportedm_ignore_confirm'] = 'Are you sure you wish to ignore further reports about this user\'s profile?\\n\\nThis will turn off further reports for everyone.';
+$txt['mc_reportedm_ignore_confirm'] = 'If you choose &quot;OK&quot;, all future reports about this user\'s profile will be ignored; moderators will not see any future reports about this user\'s profile . Are you sure you wish to do this?';
```
Comment (m4z) [inline Themes/default/languages/ModerationCenter.english.php:174]: duplicate key?

File: Themes/default/languages/EmailTemplates.english.php (hunk 1 above)
Comment (frandominguezl) [reply Themes/default/languages/EmailTemplates.english.php]: Addressed in https://github.com/SimpleMachines/SMF2.1/pull/5778/commits/b68a6dcdc21ea1244e9011f7d62c7b20b3b54c50

File: Themes/default/languages/ModerationCenter.english.php (hunk 3 above)
Comment (frandominguezl) [reply Themes/default/languages/ModerationCenter.english.php:174]: Addressed in https://github.com/SimpleMachines/SMF2.1/pull/5778/commits/b68a6dcdc21ea1244e9011f7d62c7b20b3b54c50

File: Themes/default/languages/Help.english.php (hunk 2 above)
Comment (frandominguezl) [reply Themes/default/languages/Help.english.php]: Addressed in https://github.com/SimpleMachines/SMF2.1/pull/5778/commits/b68a6dcdc21ea1244e9011f7d62c7b20b3b54c50

**Comment by frandominguezl** [comment]
I require some input with the latest commit https://github.com/SimpleMachines/SMF2.1/pull/5778/commits/410bb98e6357b5e7547a4189568eb89704ac8cc2

It kinda solved #5748, but I have a few questions. $file->fetch('utf8') should return the string associated with 'utf8' _(which actually should be 'utf8' as well)_ so the operator was wrongly set. I tried replacing it by `$file->fetch('utf8') == 'utf8' ? $txt['yes'] : $txt['no']` but no luck. Apparently $file->fetch() is not working especially for 'utf8', but I couldn't figure out why. A fast check showed me that it returns an empty string, but I just did it very quickly.

Now the thing is, it's kinda redundant to have an operator in there since all language packs in 2.1 are UTF-8 but I'm also curious on why that particular function isn't working. So, we have two options:

- Leave it as I did, by just assigning `'utf8' => $txt['yes']`
- Figure out why $file->fetch('utf8') is not working and change the operator

Any thoughts on this?

**Comment by MissAllSunday** [comment]
You can assign $file->fetch('utf8') to a var and check for empty on that var:

$var = $file->fetch('utf8')

...

'utf8' => !empty($var) ? $txt['yes'] : $txt['yes'],


$file->fetch('utf8') probably fails because https://download.simplemachines.org/fetch_language.php?version=2.1 RC2. isn't returning what we expect.

But yeah, seems redundant to have that ternary.

File: Themes/default/languages/Help.english.php [hunk 4]
```diff
@@ -356,10 +356,10 @@
...
-$helptxt['securityDisable_moderate'] = 'This <em>disables</em> the additional password check for the moderation page. This is not recommended!';
-$helptxt['securityDisable_moderate_why'] = 'This is your current password. (the same one you use to login).<br><br>The requirement to enter this helps ensure that you want to do whatever moderation you are doing, and that it is <strong>you</strong> doing it.';
+$helptxt['securityDisable'] = 'When enabled, this <em>disables</em> the additional password check for the administration page. This is not recommended!';
+$helptxt['securityDisable_why'] = 'Enter your current password. (The same one you use to login.)<br><br>The requirement to enter this helps ensure that you want to do whatever administration you are doing, and that it is <strong>you</strong> doing it.';
+$helptxt['securityDisable_moderate'] = 'When enabled, this <em>disables</em> the additional password check for the moderation page. This is not recommended!';
```
Comment (m4z) [inline Themes/default/languages/Help.english.php]:
```suggestion
$helptxt['securityDisable_moderate'] = 'When enabled, this <em>disables</em> the additional password check for the moderation page. This setting is enabled by default.';
```

File: Themes/default/languages/EmailTemplates.english.php [hunk 5]
```diff
@@ -200,7 +200,7 @@
 $txt['new_announcement_subject'] = 'New announcement: {TOPICSUBJECT}';
 $txt['new_announcement_body'] = '{MESSAGE}
 
-To unsubscribe from these announcements, login to the forum and uncheck "Receive forum announcements and important notifications by email." in your profile.
+To unsubscribe from these announcements, login to the forum and uncheck "Receive forum announcements and important notifications by email." in your profile <em>(Your username > Notifications > General Settings)</em>.
```
Comment (m4z) [inline Themes/default/languages/EmailTemplates.english.php]:
```suggestion
To unsubscribe from these announcements, login to the forum and uncheck "Receive forum newsletters, announcements and important notifications by email." in your profile <em>(Your username > Notifications > General Settings)</em>.
```
As described in `19. a)`.

File: Themes/default/languages/ManagePaid.english.php [hunk 6]
```diff
@@ -45,7 +45,7 @@
 $txt['paid_currency_code'] = 'Currency Code';
 $txt['paid_currency_code_desc'] = 'Code used by payment merchants';
 $txt['paid_currency_symbol'] = 'Symbol used by payment method';
-$txt['paid_currency_symbol_desc'] = 'Use \'%1.2f\' to specify where number goes, for example $%1.2f, %1.2fDM etc';
+$txt['paid_currency_symbol_desc'] = 'Use \'%1.2f\' to specify where number goes. For example $%1.2f, %1.2fDM etc';
```
Comment (m4z) [inline Themes/default/languages/ManagePaid.english.php:48]: Hehe, in `22 a)` I actually meant the fullstop after "etc.", but apparently, as I just learned, in english, "etc" without the dot is also valid. So nothing more to change here. 😄

**Comment by MissAllSunday** [comment]
@frandominguez03  and @m4z do you consider this as ready to go?

**Comment by m4z** [comment]
Not yet, see [here](https://github.com/SimpleMachines/SMF2.1/issues/5661#issuecomment-533676458).

**Comment by frandominguezl** [comment]
This should be ready to go. Once again, huge thanks to @m4z for putting in a lot of effort going through the language files. I really appreciate that!

**Comment by m4z** [comment]
Thank *you* for fixing it all!
(I'll report more after RC3 (and then PR it myself). 😉)

**Review by MissAllSunday — APPROVED** [review]

**Pull request merged**